src/main/java/LocalKeyClientEncryption.java

```diff
@@ -0,0 +1,180 @@
+        BlobClient blobClient = new BlobClientBuilder()
+                .endpoint(storageAccountUrl)
+                .credential(new StorageSharedKeyCredential(storageAccount, sharedKeyCred))
+                .containerName(containerName)
+                .blobName(blobName + blobSuffix)
+                .buildClient();
+
+        // Set blob encryption client
+        if (key == null) {
+            key = this.key;
+        }
+        this.setLocalKeyClient(key, blobClient);
+
+        // Downloading encrypted blob, blob is decrypted upon download
+        String fileName = blobName + "Decrypted" + blobSuffix;
+        this.encryptedBlobClient.downloadToFile(".\\src\\main\\java\\" + fileName);
+
+        // Creating blob client for decryption and reuploading
+        BlobClientBuilder blobClientBuilder = new BlobClientBuilder()
+                .endpoint(storageAccountUrl)
+                .credential(new StorageSharedKeyCredential(storageAccount, sharedKeyCred))
+                .containerName(containerName)
+                .blobName(fileName);
+
+        // Checking to see if local key is provided to use customer-managed key for server encryption
+        if (serverKey != null){
+            String CPKfileName = blobName + "DecryptedCPK" + blobSuffix;
+            blobClientBuilder = blobClientBuilder
+                    .customerProvidedKey(serverKey)
+                    .blobName(CPKfileName);
+        }
+        BlobClient blobClientDecrypted = blobClientBuilder.buildClient();
+
+        // Uploading file to server
+        blobClientDecrypted.uploadFromFile(".\\src\\main\\java\\" + fileName);
+
+        // Cleaning up by deleting local save of encrypted blob
+        File localFile = new File(".\\src\\main\\java\\" + fileName);
+        localFile.delete();
+
+    }
+
+    /**
+     * Requires setting and naming the following environmental variables (must be in system environmental variables):
+     * sharedKeyCred (for storage account)
+     * storageAccount
+     * This sample will show client-side decryption using a local key, and then upload two files with server-side
+     * encryption. One will be encrypted with Microsoft-managed keys, and the other is by customer-provided keys.
+     */
+    public static void main(String[] args){
+        String sharedKeyCred = System.getenv("sharedKeyCred");
+        String storageAccount = System.getenv("storageAccount");
+
+        // Setting names of container and blob so that they match ones from setup. Can be changed in case where
+        // set up is not used
+
+        String containerName = "containername";
+        String blobName = "blobExample";
+        String blobSuffix = ".txt";
+
+        // Creating LocalKeyClientEncryption helper
+        LocalKeyClientEncryption helper = new LocalKeyClientEncryption();
+        // Setup where sample blob is client-side encrypted and uploaded to server
+        helper.setup(storageAccount, sharedKeyCred);
```

Comment:
To prevent needing to make the helper object you could just make everything static